go-controller/pkg/retry/obj_retry.go

```diff
@@ -288,9 +290,13 @@ func (r *RetryFramework) resourceRetry(objKey string, now time.Time) {
 		if r.ResourceHandler.NeedsUpdateDuringRetry && entry.config != nil && entry.newObj != nil {
 			klog.Infof("%v retry: updating object %s", r.ResourceHandler.ObjType, objKey)
 			if err := r.ResourceHandler.UpdateResource(entry.config, entry.newObj, true); err != nil {
-				klog.Infof("%v retry update failed for %s, will try again later: %v", r.ResourceHandler.ObjType, objKey, err)
 				entry.timeStamp = time.Now()
 				entry.failedAttempts++
+				if entry.failedAttempts >= MaxFailedAttempts {
+					klog.Errorf("Retry update failed final attempt for %s %s: error: %v", r.ResourceHandler.ObjType, objKey, err)
+				} else {
+					klog.Infof("%v retry update failed for %s, will try again later: %v", r.ResourceHandler.ObjType, objKey, err)
+				}
```

Comment:
Does it make sense to have a different behavior in `WatchResourceFiltered` vs `resourceRetry`?

Here we are just basing our log on the attempts and not on the the type of error. 

Is the general intention to log error on the first and last attempt, and log info for attempts in between, unless it is a SuppressedError for which the first attempt is log infoas well?

Could we benefit from a `handleErrLog(attempt int, err error)` method that we use everywhere so that this is easier to understand what we intend to do?

Reply:
the intention is to log info on certain errors that we dont expect to work on first attempt, but if they exceed max retries and never work, then we log an error

Reply:
oh I just realized I missed checking if error was suppressed type here..

Reply:
we agreed to defer this to later